From fc728ed7c9aee5cb9129adc7d373c6541e34f99d Mon Sep 17 00:00:00 2001
From: keyur-mpc <keyur@mpconsultants.org>
Date: Tue, 1 Aug 2023 15:43:02 +0000
Subject: [PATCH 04/14] Configured pins to be used as GPIO Disabled node which
 was using the same PAD for pins to be used as GPIO

---
 ...x6ull-var-som-concerto-board-emmc-wifi.dts | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
index 56d41eef3234..0a2ae49ab708 100644
--- a/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
+++ b/arch/arm/boot/dts/imx6ull-var-som-concerto-board-emmc-wifi.dts
@@ -69,6 +69,25 @@
 			MX6UL_PAD_GPIO1_IO01__GPIO1_IO01 0xb0
 		>;
 	};
+
+    pinctrl_java_gpio: javagpiogrp {
+		fsl,pins = <
+			MX6UL_PAD_LCD_DATA04__GPIO3_IO09 0x1b0b0
+			MX6UL_PAD_LCD_DATA17__GPIO3_IO22 0x1b0b0
+			MX6UL_PAD_LCD_DATA03__GPIO3_IO08 0x1b0b0
+			MX6UL_PAD_LCD_DATA12__GPIO3_IO17 0x1b0b0
+			MX6UL_PAD_LCD_DATA20__GPIO3_IO25 0x1b0b0
+			MX6UL_PAD_LCD_DATA13__GPIO3_IO18 0x1b0b0
+			MX6UL_PAD_GPIO1_IO02__GPIO1_IO02 0x1b0b0
+			MX6UL_PAD_UART5_RX_DATA__GPIO1_IO31 0x1b0b0
+			MX6ULL_PAD_SNVS_TAMPER9__GPIO5_IO09 0x1b0b0
+			MX6UL_PAD_LCD_DATA02__GPIO3_IO07 0x1b0b0
+			MX6UL_PAD_SNVS_TAMPER7__GPIO5_IO07 0x1b0b0
+			MX6UL_PAD_SNVS_TAMPER3__GPIO5_IO03 0x1b0b0
+			MX6UL_PAD_CSI_DATA02__GPIO4_IO23 0x1b0b0
+			MX6UL_PAD_GPIO1_IO05__GPIO1_IO05 0x1b0b0
+		>;
+	};
 };
 
 /* wdog to avoid confclicts with ADC Pin */
@@ -114,3 +133,18 @@
 	num-channels = <10>;
 	status = "okay";
 };
+
+/* Disabled to avoid conflicts with GPIO3_IO07, GPIO3_IO09, GPIO3_IO08, GPIO3_IO17, GPIO3_IO25, GPIO3_IO18 */
+&lcdif {
+    status = "disabled";
+};
+
+/* Disabled to avoid conflicts with GPIO1_IO05 */
+&pwm4 {
+    status = "disabled";
+};
+
+/* Disabled to avoid conflicts with GPIO1_IO02 */
+&tsc {
+    status = "disabled";
+};
-- 
2.25.1

